proto/site/v1alpha/site.proto

```diff
@@ -204,4 +213,57 @@ message Member {
 // TODO: define the actual /.well-known path.
 message DiscoveryConfig {
   // TODO: add the necessary fields to export under the /.well-known HTTP discovery path.
-}
\ No newline at end of file
+}
+
+// Request the server to pin+publish a document.
+message PublishRequest {
+  // id of the doc to publish
+  string doc_id = 1;
+
+  // (optional) pretty path to publish at. Empty string === home doc.
+  string path = 2;
+}
+
+// Response when publishing a document.
+message PublishResponse {
+  // id of the web publication on this server
+  string publication_id = 1;
+}
+
+// Request Unpin/Unpublish a document from the site.
+message UnpublishRequest {
+  // id of web publication on this server to unpublish
+  string publication_id   = 1;
+}
+
+message UnpublishResponse {}
+
+// Object representing a "web publication" row on this server, including fields for user presentation (doc_title, author_name, update_time)
+message ListedWebPublication {
+  // id of web publication on this server
+  string publication_id = 1;
```

Comment:
I have the same question @ericvicenti

Reply:
what if the same document is published twice to different paths? it would be two publications, right?

happy to just use doc_id if you prefer to simplify it

Reply:
If you just want to change the pretty Path only, you can publish the same document changing that field and when listed here in `ListedWebPublication` there will be two elements (since `ListWebPublicationsResponse` is a list) one with a path different than the other. Since the path is optional, when not provided, that will be root `/` and there can't be two documents in root so the second publish gRPC call would fail

Reply:
when a path is not provided (null/undefined), it indicates the document is unlisted, not a home doc.

the home doc was an empty string path at one point but it could also be hardcoded into the site